phoenix-core/src/main/java/org/apache/phoenix/compile/ProjectionCompiler.java

```diff
@@ -505,7 +505,15 @@ public static RowProjector compile(StatementContext context, SelectStatement sta
         } else {
             isProjectEmptyKeyValue = where == null || LiteralExpression.isTrue(where) || where.requiresFinalEvaluation();
             for (byte[] family : projectedFamilies) {
-                projectColumnFamily(table, scan, family);
+                try {
+                    if (table.getColumnFamily(family) != null) {
+                        projectColumnFamily(table, scan, family);
+                    }
+                } catch (ColumnFamilyNotFoundException e) {
+                    if (!(tableRef.getTable().getIndexType() == IndexType.LOCAL || isHintedGlobalIndex(tableRef))) {
+                        throw e;
+                    }
```

Comment:
for else case, would a DEBUG log be helpful?

Reply:
This is again synced from 5.2, which does not have a debug statement there.
This applies to all server merged columns, so it would generate a log line for each merged column, which sounds excessive.